ADPluginKafkaApp/src/NDArraySerializer.cpp

```diff
@@ -53,34 +53,35 @@ void NDArraySerializer::SerializeData(NDArray &pArray,
   // Get all attributes of this data package
   std::vector<flatbuffers::Offset<Attribute>> attrVec;
 
-  // When passing NULL, get first element
-  NDAttribute *attr_ptr = pArray.pAttributeList->next(nullptr);
-
-  // Itterate over attributes, next(ptr) returns NULL when there are no more
-  while (attr_ptr != nullptr) {
-    auto temp_attr_str = builder.CreateString(attr_ptr->getName());
-    auto temp_attr_desc = builder.CreateString(attr_ptr->getDescription());
-    auto temp_attr_src = builder.CreateString(attr_ptr->getSource());
+  // Iterate over attributes, next(ptr) returns NULL when there are no more
+  for (NDAttribute *attr_ptr = pArray.pAttributeList->next(nullptr); attr_ptr != nullptr; attr_ptr = pArray.pAttributeList->next(attr_ptr)) {
     size_t bytes;
     NDAttrDataType_t c_type;
-    attr_ptr->getValueInfo(&c_type, &bytes);
+    int result = attr_ptr->getValueInfo(&c_type, &bytes);
+    if ((ND_SUCCESS != result) | (0 == bytes))
```

Comment:
I compared `c_type` against `NDAttrUndefined` instead of comparing `bytes` against `0`.

They're equivalent of course, but you need to inspect the implementation of `getValueInfo` to know that. The condition that we actually care about is that `c_type` is not `NDAttrUndefined`, so that we know the switch statement is exhaustive.